File: TaskCRUD/Models/Tasks.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.ComponentModel.DataAnnotations;
using System.Runtime.Serialization;

namespace TaskCRUD.Models
{
    public class Tasks
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }
        [Required]
        public string Name { get; set; }
        public string? Description { get; set; }
        public DateTime DueDate { get; set; }
        public int StatusId { get; set; }
    }
}

File: UI/TaskManager/src/app/pages/Models/Tasks.ts
export interface Tasks{
    id?:string;
    name:string;
    description:string;
    duedate: Date; 

    // constructor(id: string, name: string, description: string, duedate : Date) {
    //     this.id = id;
    //     this.name = name;
    //     this.description = description;
    //     this.duedate=duedate;
    // }
}